src/constraints/storage_constraints.jl

```diff
@@ -81,13 +81,17 @@ function add_elec_storage_dispatch_constraints(m, p, b; _n="")
             + p.s.storage.attr[b].grid_charge_efficiency * m[Symbol("dvGridToStorage"*_n)][b, ts] 
             - m[Symbol("dvDischargeFromStorage"*_n)][b,ts] / p.s.storage.attr[b].discharge_efficiency
         )
+        - (p.s.storage.attr[b].soc_based_per_ts_self_discharge_fraction * m[Symbol("dvStoredEnergy"*_n)][b, ts])
+        - (p.s.storage.attr[b].capacity_based_per_ts_self_discharge_fraction * m[Symbol("dvStorageEnergy"*_n)][b])
```

Comment:
@lixiangk1 based on our convo last week - this "per_ts" method was also going to be adopted for H2, right? (instead of "daily leakage"). I agree with that.